src/component/mxgraph/shape/render/IconPainter.ts

```diff
@@ -311,15 +311,32 @@ export default class IconPainter {
     canvas.fillAndStroke();
   }
 
+  private static drawCrossIcon(canvas: BpmnCanvas): void {
+    canvas.begin();
+    canvas.moveTo(0.38, 0);
+    canvas.lineTo(0.62, 0);
+    canvas.lineTo(0.62, 0.38);
+    canvas.lineTo(1, 0.38);
+    canvas.lineTo(1, 0.62);
+    canvas.lineTo(0.62, 0.62);
+    canvas.lineTo(0.62, 1);
+    canvas.lineTo(0.38, 1);
+    canvas.lineTo(0.38, 0.62);
+    canvas.lineTo(0, 0.62);
+    canvas.lineTo(0, 0.38);
+    canvas.lineTo(0.38, 0.38);
+    canvas.close();
+  }
+
   /**
    * This icon is used by `exclusive gateway`.
    */
-  public paintXCrossIcon({ c, ratioFromParent, setIconOrigin, shape, icon }: PaintParameter): void {
-    const canvas = this.newBpmnCanvas({ c, ratioFromParent, setIconOrigin, shape, icon: { ...icon, isFilled: true } }, { height: 0.5, width: 0.5 });
+  public paintXCrossIcon(paintParameter: PaintParameter): void {
+    const canvas = this.newBpmnCanvas(paintParameter, { height: 0.5, width: 0.5 });
 
     IconPainter.drawCrossIcon(canvas);
-    const rotationCenterX = shape.w / 4;
```

Comment:
If someone change the ratioFromParent (the size of the icon from the size of the shape), we always want to rotate the icon with the same value.

Can you revert this change, please ?

Reply:
Forget my previous comment.
I'm not sur to understand how it's works, but everthing is OK. 

Thank you very much for having integrate my comments 😄

Reply:
It's the calculation of coordinates of rotation center.
Previously it was hardcoded to 1/4th cause of initial size of icon (half-size of shape) => center of rotation was on 1/4th of shape, but must be proportional to the ratioFromParent.
BTW, the default **ratioFromParent** is equals to 0.25 ;)